src/LDLFactorizations.jl

```diff
@@ -483,10 +486,14 @@ function ldl_factorize!(A::Symmetric{T,SparseMatrixCSC{T,Ti}},
 
   # perform numerical factorization
   S.__factorized = false
-  ldl_numeric_upper!(S.n, A.data.colptr, A.data.rowval, A.data.nzval,
-                     S.Cp, S.Ci, S.Lp, S.parent, S.Lnz, S.Li, S.Lx, S.d, S.Y, S.pattern, S.flag, S.P, S.pinv,
-                     S.r1, S.r2, S.tol, S.n_d)
-  S.__factorized = true
+  out = ldl_numeric_upper!(S.n, A.data.colptr, A.data.rowval, A.data.nzval,
+                           S.Cp, S.Ci, S.Lp, S.parent, S.Lnz, S.Li, S.Lx, S.d, S.Y, S.pattern, S.flag, S.P, S.pinv,
+                           S.r1, S.r2, S.tol, S.n_d)
+  if out == 0 
+    S.__factorized = true
+  elseif out == 1 && S.avoid_break == false
```

Comment:
I suggest to remove `avoid_break` and have `ldl_numeric_upper!()` return a bool to we can write
```julia
S.__factorized = ldl_numeric_upper!(...)
```

Reply:
If we remove `avoid_break` then we cannot raise an error by default (which was the previous behaviour) when factorization fails

Reply:
But maybe, `S.__factorized` to `false` is sufficient to say that the factorization failed, and we don't need `avoid_break`?

Reply:
Yes, but I think that for users that are not aware of this behaviour, it makes their code hard to debug, because their algorithm may continue with NaN or other similar issues. Even though using `ldiv!` will raise an error, it is not explicit that it comes from an issue within the factorization.
It's just my point of view though, if you both want to remove the errors when the factorization fails then I'll do it (I will also have to change the other `ldl_factorize!` function L558).

Reply:
What was the motivation for this change in the first place? To avoid using `try ... catch` in your code?

Reply:
yes

Reply:
We already have `factorized()`, so we just need to document that it’s recommended to check that a factorization succeeded before calling `ldiv!()`. Also, the error message in `ldiv!()` can be more explicit.